server/src/models/errors/api-error.ts

```diff
@@ -1,29 +1,42 @@
 import { LogService } from "./../../logic/services/index";
+import { ERROR_MESSAGES } from "../../error-messages"
+import { ERRORS } from "../../errors"
+import { ErrorType } from "./../index";
 
-
-export class ApiError {
-  private timestamp: number;
-  private name: string;
-  private stackTrace: string;
-  private message: string;
-  private source: string;
-  private sourceMethod: string;
+export class ApiError extends Error{
+  timestamp: number;
+  statusCode: number;
+  code: string;
+  message: string;
+  name = ApiError.name;
+  source = "";
+  sourceMethod = "";
+  error = new Error();
 
   constructor (
-    message: string, source: string, 
-    sourceMethod: string, error: any
+    errorType: ErrorType
   ) {
+    super();
+    this.message = this.resolveMessage(
+      errorType.messageKey, 
+      errorType.messageArgs);
+    this.statusCode = errorType.statusCode;
+    this.code = errorType.code;
     this.timestamp = Date.now();
-    this.name = error.name;
-    this.stackTrace = error.stack;
-    this.message = message;
-    this.source = source;
-    this.sourceMethod = sourceMethod;
-    // add error to Log Object
-    LogService.addError(this.message, this);
   }
 
-  forResponse(): Object {
+  private resolveMessage = (messageKey: string, args: string[]): string => {
+    let msg: string = ERROR_MESSAGES[messageKey];
+    let argCounter = 0
+    for (let arg of args) {
+      msg = msg.replace(`{${argCounter}}`, arg);
+      argCounter ++;
+    }
+    return msg;
+  }
+
+  forResponse = (): Object => {
```

Comment:
nice catch. only this function declaration works. 👍  the old forResponse(): Object {..} function declaration throws not found errros!